sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/AstyanaxDataReaderDAO.java

```diff
@@ -1112,20 +1162,18 @@ private Record emptyRecord(Key key) {
     }
 
     private Iterator<Change> decodeColumns(Iterator<Column<UUID>> iter) {
-        return Iterators.transform(iter,
-                new Function<Column<UUID>, Change>() {
-                    @Override
-                    public Change apply(Column<UUID> column) {
-                        return _changeEncoder.decodeChange(column.getName(), column.getByteBufferValue());
-                    }
-                });
+        return Iterators.transform(iter, column -> _changeEncoder.decodeChange(column.getName(), column.getByteBufferValue()));
+    }
+
+    private Iterator<Change> decodeDeltaColumns(Iterator<Column<UUID>> iter) {
+        return Iterators.transform(iter, column -> _changeEncoder.decodeChange(column.getName(), removePrefix(column.getByteBufferValue(), _deltaPrefixLength)));
```

Comment:
As we spoke about, I don't think this is worth think this is worth, as it would force us to create a new StitchedRow for every delta, even ones that are already a single block